src/ensembl/src/content/app/browser/track-panel/trackPanelReducer.ts

```diff
@@ -34,9 +35,18 @@ export default function trackPanel(
   switch (action.type) {
     case getType(browserActions.setDataFromUrl):
       const { activeGenomeId } = action.payload;
+      const newTrackPanelStateForGenome = state[activeGenomeId]
+        ? {
+            ...state[activeGenomeId],
+            ...getPersistentTrackPanelStateForGenome(activeGenomeId)
+          }
+        : {
+            ...getInitialTrackPanelStateForGenome(activeGenomeId)
+          };
```

Comment:
I'm confused. Why do we even need to read the data from the browser storage, when it should already be available in the redux state? Shouldn't we rather do the following:

```
if (!state[activeGenomeId]) {
  return { ...state, [activeGenomeId]: getTrackPanelStateForGenome(activeGenomeId)  }
} else {
  return state;
}
```

Reply:
Changed the code to be like this in 19202e8